packages/tenant-process/test/api/addCertifiedAttribute.test.ts

```
@@ -0,0 +1,111 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { describe, it, expect, vi } from "vitest";
+import request from "supertest";
+import jwt from "jsonwebtoken";
+import { generateId, Tenant } from "pagopa-interop-models";
+import {
+  createPayload,
+  getMockAuthData,
+  getMockTenant,
+} from "pagopa-interop-commons-test";
+import { userRoles, UserRole } from "pagopa-interop-commons";
+import { tenantApi } from "pagopa-interop-api-clients";
+import { api } from "../vitest.api.setup.js";
+import { tenantService } from "../../src/routers/TenantRouter.js";
+import { toApiTenant } from "../../src/model/domain/apiConverter.js";
+import {
+  attributeDoesNotBelongToCertifier,
+  attributeNotFound,
+  certifiedAttributeAlreadyAssigned,
+  tenantIsNotACertifier,
+  tenantNotFound,
+} from "../../src/model/domain/errors.js";
+
+describe("API /tenants/{tenantId}/attributes/certified authorization test", () => {
+  const tenantAttributeSeed: tenantApi.CertifiedTenantAttributeSeed = {
+    id: generateId(),
+  };
+  const tenant: Tenant = getMockTenant();
+
+  const apiResponse = tenantApi.Tenant.parse(toApiTenant(tenant));
```

I'm basing this on #1645. I'm not sure if we should be more explicit when testing the API response, that is, building the expected object explicitly as a literal instead of using the converter and schema. Since the converter and schema are the same as used in the router, if by mistake we changed the converter in a wrong way the `expect(res.body).toEqual(apiResponse)` assertion would still pass.

But if I understand correctly you were thinking of maybe testing converters separately as unit tests? Is that already done or tracked?

---

This point is still under discussion, so I can't give you a definite answer.
But the idea was to test the converters just as they happen in the router.
Also because if I understood correctly what you mean, if you do a wrong conversion (for example the `apiResponse` has onboardedAt as a date and not as a string) the expect with the `res.body` goes into error.
Or if you do the wrong conversion of the object,
For example 
`tenantApi.Tenant.parse(tenant);`
instead of `tenantApi.Tenant.parse(toApiTenant(tenant));`
It should return a parsing error.


If I haven't answered your question fully, let's talk.